mypy/checkexpr.py

```diff
@@ -624,7 +623,7 @@ def check_typeddict_call(
             item_names = cast(List[str], arg_names)
             item_args = args
             return self.check_typeddict_call_with_kwargs(
-                callee, OrderedDict(zip(item_names, item_args)), context
+                callee, dict(zip(item_names, item_args)), context
```

Comment:
Dict ordering is an implementation detail in 3.6, so I'm not sure if we should do this. Practically it makes sense, but maybe it is a bad idea?

Reply:
We anyway rely on it everywhere. If you look at backports.py, `OrderdDict` was just alias for `dict`.

Reply:
Also it is part of the spec since 3.7, see [the ruling](https://mail.python.org/pipermail/python-dev/2017-December/151283.html)